### PR TITLE
move headers to middleware

### PR DESCRIPTION
**Pull request by ksvirkou-hubspot**

### CODE REVIEW

**Comment by ksvirkou-hubspot** [comment]
https://github.com/HubSpot/hubspot-api-nodejs/issues/76

**Comment by atanasiuk-hubspot** [comment]
Lets extract `BaseDiscovery` methods to a separate class `ApiClientConfirator` or similar

File: src/discovery/IRequestContext.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface IRequestContext {
```
Comment (atanasiuk-hubspot) [inline src/discovery/IRequestContext.ts:1]: Please extract to a separate module

File: src/discovery/Observable.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export class Observable<T> {
```
Comment (atanasiuk-hubspot) [inline src/discovery/Observable.ts:1]: Please also extract from `discovery` module

**Comment by ksvirkou-hubspot** [comment]
Fixed

File: src/configuration/ApiClientConfirator.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { IConfiguration } from '../configuration/IConfiguration'
+import { VERSION } from '../configuration/version'
+import { IRequestContext } from '../services/IRequestContext'
+import { Observable } from '../services/Observable'
+
+export class ApiClientConfirator {
```
Comment (atanasiuk-hubspot) [inline src/configuration/ApiClientConfirator.ts:6]: Please fix a type in `ConfiGUrator`

File: src/configuration/ApiClientConfirator.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          hapikey: config.developerApiKey,
+        },
+        middleware: this.getHeaderMiddleware<RequestContextType, ResponseContextType>(),
+      }
+    }
+
+    return params
+  }
+
+  public static getHeaderMiddleware<RequestContextType extends IRequestContext, ResponseContextType>() {
```
Comment (atanasiuk-hubspot) [inline src/configuration/ApiClientConfirator.ts:37]: should this method be private/protected or do we call this somewhere outside?

**Review by atanasiuk-hubspot — APPROVED** [review]
🚀

**Pull request merged**